ynot93/tele-thrive
# Update and Delete Feature for Appointments

# Update and Delete Feature for Appointments

## Description
This pull request adds functionality to update and delete appointments in the TeleThrive application.

## Changes Made
- Added routes for editing and deleting appointments.
- Created templates for editing appointments and confirming deletion with a modal.
- Implemented forms and validation for updating appointments.
- Integrated JavaScript to handle modal behaviour and update delete links dynamically.

## Screenshots
![Screenshot 2024-04-15 at 10 27 40](https://github.com/ynot93/tele-thrive/assets/26874722/955766b3-20f9-4492-bb4d-2ea694b2c8b9)
![Screenshot 2024-04-15 at 10 27 58](https://github.com/ynot93/tele-thrive/assets/26874722/ec7e17f9-afbd-416c-b35b-60a1223e5cbf)


## How to Test
1. Navigate to the appointments page in the dashboard.
2. Click on the edit button for an existing appointment.
3. Verify that the edit form is displayed with pre-filled data.
4. Make changes to the appointment details and submit the form.
5. Check that the appointment is updated successfully.
6. Click on the delete button for an existing appointment.
7. Confirm the deletion in the modal.
8. Verify that the appointment is deleted.

## Checklist
- [x] Implemented update appointment route and form
- [x] Implemented delete appointment route and confirmation modal
- [x] Added appropriate validation for update and delete actions
- [x] Tested the functionality thoroughly
- [x] Updated documentation and comments as needed
